bigbone/src/main/kotlin/social/bigbone/api/method/StreamingMethods.kt

```diff
@@ -1,279 +1,207 @@
                 }
-                reader.close()
-            }
-            return Shutdownable(dispatcher)
-        } else {
-            throw BigBoneRequestException(response)
-        }
-    }
 
-    @Throws(BigBoneRequestException::class)
-    fun localPublic(handler: Handler): Shutdownable {
-        val response = client.get("api/v1/streaming/public/local")
-        if (response.isSuccessful) {
-            val reader = response.body?.byteStream()?.bufferedReader()
-            val dispatcher = Dispatcher()
-            dispatcher.invokeLater {
-                while (true) {
-                    try {
-                        val line = reader?.readLine()
-                        if (line == null || line.isEmpty()) {
-                            continue
-                        }
-                        val type = line.split(":")[0].trim()
-                        if (type != "event") {
-                            continue
-                        }
-                        val event = line.split(":")[1].trim()
-                        val payload = reader.readLine()
-                        val payloadType = payload.split(":")[0].trim()
-                        if (payloadType != "data") {
-                            continue
-                        }
-                        if (event == "update") {
-                            val json = payload.substringAfter(':').trim()
-                            handler.onStatus(status = JSON_SERIALIZER.decodeFromString(json))
-                        }
-                    } catch (e: java.io.InterruptedIOException) {
-                        break
+                if (streamType == HASHTAG || streamType == HASHTAG_LOCAL) {
+                    requireNotNull(tagName) {
+                        "When requesting $streamType for stream, a non-null tag name needs to be specified"
                     }
+                    append("tag", tagName)
                 }
-                reader.close()
-            }
-            return Shutdownable(dispatcher)
-        } else {
-            throw BigBoneRequestException(response)
+            },
+            callback = callback
+        )
+
+        return Closeable {
+            webSocket.close(
+                /*
+                1000 indicates a normal closure,
+                meaning that the purpose for which the connection was established has been fulfilled.
+                see: https://datatracker.ietf.org/doc/html/rfc6455#section-7.4
+                 */
+                code = 1000,
+                reason = null
+            )
         }
     }
 
-    @Throws(BigBoneRequestException::class)
-    fun federatedHashtag(tag: String, handler: Handler): Shutdownable {
-        val response = client.get(
-            "api/v1/streaming/hashtag",
-            Parameters().append("tag", tag)
+    /**
+     * Verify that the streaming service is alive before connecting to it.
+     * Returns "OK" if the streaming service is alive.
```

Comment:
The [API documentation](https://docs.joinmastodon.org/methods/streaming/#health) unfortunately doesn’t mention at all what would happen if it were “unhealthy”, so the only good idea I had was to turn this into an error.

Should we find out what would be returned in case it’s not “OK”, I’d be happy to find a better approach!

I’ll update documentation here to add the `onError` return info.

Reply:
Done. ✅